rider/src/test/kotlin/ConnectionTest.kt

```diff
@@ -0,0 +1,52 @@
+import com.intellij.util.io.exists
+import com.jetbrains.rd.util.reactive.hasTrueValue
+import com.jetbrains.rdclient.util.idea.waitAndPump
+import com.jetbrains.rider.plugins.unity.UnityHost
+import com.jetbrains.rider.plugins.unity.actions.StartUnityAction
+import com.jetbrains.rider.test.base.BaseTestWithSolution
+import com.jetbrains.rider.test.framework.combine
+import com.jetbrains.rider.test.framework.downloadAndExtractArchiveArtifactIntoPersistentCache
+import com.jetbrains.rider.test.scriptingApi.changeFileSystem2
+import com.jetbrains.rider.test.scriptingApi.checkSwea
+import com.jetbrains.rider.util.idea.lifetime
+import org.testng.annotations.Test
+import java.io.File
+import java.nio.file.Paths
+import kotlin.test.assertNotNull
+
+class ConnectionTest : BaseTestWithSolution() {
+    override fun getSolutionDirectoryName(): String {
+        return "SimpleUnityProject"
+    }
+
+    override val waitForCaches = true;
+
+    var unityPackedUrl = "https://repo.labs.intellij.net/dotnet-rider-test-data/Unity_2018.3.4f1_stripped_v4.zip";
```

Comment:
I don't like that we have to depend on a repackaged Unity on a private URL. Means no-one external to the company can run this test, but I don't really know what the alternative is.

Reply:
Perhaps use installation finder to try and find 2018.3 locally, and if not available, download and use this?

Reply:
This tool might help, too: https://github.com/DragonBox/u3d